### DISCUSSION TITLE
Bayesian Calibration with Mixed Field/Scalar Calibration Data

### DISCUSSION THREAD

**Discussion by jeanstgo** [snl-dakota/dakota#152] [Q&A] [answered]
<img width="437" alt="dakota_input_bayes_cal" src="https://github.com/user-attachments/assets/3c5874c8-1a1f-4dd2-857d-d62436aefea7">

I have the following input deck setup to run a Bayesian calibration using QUESO. When I try to run the study, I get the error: "number of levels not evenly divisible by the number of response functions".  What "levels" is the message referring to and how can I fix it? I would also like to know if the response block is set up correctly to take in mixed field/scalar calibration terms. 

Thanks in advance

**Comment by thenectarine** [comment, marked as answer]
I am still new to Dakota, but I have seen the same error often.

Your responses keyword block is configured for N responses. I don't know the exact number of responses since I am not familiar with the calibration keywords.

Say your responses keyword block is configured for 4 responses r1, r2, r3, r4.

If your probability_levels keyword is equal to `0.05   0.1   0.05    0.1`, it is implied that the first 0.05 is associated with r1, the second value of 0.1 is associated with r2, the third value of 0.05 is associated with r3, and the fourth value of 0.1 is associated with r4. In this case, 4 probability levels is divisible by 4 responses. So, the allocation is done where each 1 probability level maps to 1 response. If there were 20 probability levels and 4 responses, every 5 probability levels is mapped to a response.

Now, what happens if you have 5 responses r1, r2, r3, r4, r5 and the same probability levels? Since 4 probability levels are not divisible by 5 responses, Dakota cannot automatically associate each probability level to each response. You can use the keyword `num_probability_levels` to explicitly state which probability levels belong to which response. For the same 5 responses and 4 probability levels, consider this.

```
probability_levels   0.05    0.1     0.05     0.1
    num_probability_levels    0     1     1     1    1
```
    
This is read as the first zero probability levels are associated to response r1, the next 1 probability levels are associated with response r2, the next 1 probability levels are associated with response r3, the next 1 probability levels are associated with response r4, and the next 1 probability levels are associated with response r5.

The Dakota Reference Manual's entry on `response_levels` provides additional information on how these level keywords operate. The Dakota Reference Manual is available in a ZIP file at this link: https://dakota.sandia.gov/manuals/. Extract the ZIP and open the index.html file in a web browser.

**Reply by jeanstgo** [reply]
Thank you, I fixed the issue by only including one probability level at a time. Analysis is running properly.
